services/worker/src/worker/job_runners/dataset/hub_cache.py

```diff
@@ -70,7 +70,8 @@ def compute_hub_cache_response(dataset: str) -> tuple[DatasetHubCacheResponse, f
             kind="dataset-compatible-libraries", dataset=dataset
         )
         tags = compatible_libraries_response["content"]["tags"]
-        libraries = compatible_libraries_response["content"]["libraries"]
+        compatible_libraries: list[CompatibleLibrary] = compatible_libraries_response["content"]["libraries"]
```

Comment:
interesting. It means that we are passing the complete code samples, etc, to the Hub through SSE, and the Hub can cache them in its own DB, removing the need to call datasets-server on every dataset page creation.

Hmmm, I don't know at which point we want to do this though. We don't want to replicate all the cache-entries database to the Hub's one.

Reply:
yes we want to only store the minimum amount of data in the Hub DB

it's important DX-wise on the Hub side 🙏 
🙏 

calling datasets-server on dataset pages is fine

Reply:
OK. For now it's sent in the SSE, but not stored on the Hub side, so it's only some additional traffic. Once we finish the "Use in libraries" thing, I'll remove the bits we don't use from this endpoint.

Reply:
I opened an issue for this: https://github.com/huggingface/datasets-server/issues/2618

Reply:
We are no longer passing the code samples btw, that's the point of this PR

Reply:
You're right, I misread